elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/Schema.java

```diff
@@ -0,0 +1,290 @@
+     * Constructor
+     * <p>
+     * This constructor calls {@link #constructDimension(String, Class, EntityDictionary)} and
+     * {@link #constructMetric(String, Class, EntityDictionary)} ()} to construct all {@link Dimension}s and
+     * {@link Metric}s associated with the entity class passed in.
+     *
+     * @param cls  The type of the entity, whose {@link Schema} is to be constructed
+     * @param entityDictionary  The meta info object that helps to construct {@link Schema}
+     *
+     * @throws NullPointerException if anyone of the arguments is {@code null}
+     */
+    public Schema(Class<?> cls, EntityDictionary entityDictionary) {
+        this.entityClass = Objects.requireNonNull(cls, "cls");
+        this.entityDictionary = Objects.requireNonNull(entityDictionary, "entityDictionary");
+
+        this.metrics = getAllMetrics();
+        this.dimensions = getAllDimensions();
+    }
+
+    /**
+     * Finds the {@link Dimension} by name.
+     *
+     * @param dimensionName  The entity field name associated with the searched {@link Dimension}
+     *
+     * @return {@link Dimension} found or {@code null} if not found
+     */
+    public Dimension getDimension(String dimensionName) {
+        return getDimensions().stream()
+                .filter(dimension -> dimension.getName().equals(dimensionName))
+                .findAny()
+                .orElse(null);
+    }
+
+    /**
+     * Finds the {@link Metric} by name.
+     *
+     * @param metricName  The entity field name associated with the searched {@link Metric}
+     *
+     * @return {@link Metric} found or {@code null} if not found
+     */
+    public Metric getMetric(String metricName) {
+        return getMetrics().stream()
+                .filter(metric -> metric.getName().equals(metricName))
+                .findAny()
+                .orElse(null);
+    }
+
+    /**
+     * Returns whether or not an entity field is a metric field.
+     * <p>
+     * A field is a metric field iff that field is annotated by at least one of
+     * <ol>
+     *     <li> {@link MetricAggregation}
+     *     <li> {@link MetricComputation}
+     * </ol>
+     *
+     * @param fieldName  The entity field
+     *
+     * @return {@code true} if the field is a metric field
+     */
+    public boolean isMetricField(String fieldName) {
+        return getEntityDictionary().attributeOrRelationAnnotationExists(
+                getEntityClass(), fieldName, MetricAggregation.class
+        )
+                || getEntityDictionary().attributeOrRelationAnnotationExists(
+                getEntityClass(), fieldName, MetricComputation.class
+        );
+    }
+
+    /**
+     * Returns whether or not an entity field is a simple metric, i.e. a metric field not decorated by
+     * {@link MetricComputation}.
+     *
+     * @param fieldName  The entity field
+     *
+     * @return {@code true} if the field is a simple metric
+     */
+    public boolean isBaseMetric(String fieldName) {
```

Comment:
Oops. Sorry I didn't remember to refactor entityDictionary out of this method. Yeah I remember you said it is costly to do reflection on this instance method.

Reply:
I don't think my earlier comment was clear.  I don't think we need this method.  We just need a way to determine if a metric has aggregations or not.  There should be a function on Metric (the interface) called `getAggregations`.  This can return an empty list.